chain/pool/txlist.go

```diff
@@ -82,25 +92,36 @@ func (nst *NonceSortedTxs) Push(tx *transaction.Transaction) error {
 
 	hash := tx.Hash()
 	nst.idx = append(nst.idx, hash)
+	if nst.txs == nil {
+		nst.txs = make(map[common.Uint256]*transaction.Transaction)
+	}
 	nst.txs[hash] = tx
 
 	return nil
 }
 
-func (nst *NonceSortedTxs) Pop() (*transaction.Transaction, error) {
+func (nst *NonceSortedTxs) PopN(n uint16) ([]*transaction.Transaction, error) {
 	nst.mu.Lock()
 	defer nst.mu.Unlock()
 
 	if nst.empty() {
-		return nil, ErrNonceSortedTxsEmpty
+		return []*transaction.Transaction{}, ErrNonceSortedTxsEmpty
 	}
 
-	hash := nst.idx[0]
-	nst.idx = nst.idx[1:]
-	tx := nst.txs[hash]
-	delete(nst.txs, hash)
+	ret := make([]*transaction.Transaction, 0)
```

Comment:
It's more efficient to use `make([]*transaction.Transaction, 0, n)` here

Reply:
Done in additional commit.